### PR TITLE
Remove device use case

### LINKED ISSUE #9: Use case for deleting a device
Devices are those who can trigger the use cases related with reminders, those could be an NFC tag, bluetooth device, etc, etc, etc.

This use case should:
1) Given the deviceId, search if the device exists (depends on #8 ), if not throw an exception (NoSuchDeviceException)
2) If the device exists, remove it from the repository.

### PR DESCRIPTION
**Pull request by oriolagobat**
Closes #9

### CODE REVIEW

File: src/main/kotlin/device/application/DeviceRemover.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package device.application
+
+import device.domain.DeviceRepository
+import device.domain.exceptions.NonExistentDeviceException
+import shared.domain.Identifier
+
+class DeviceRemover(
+    private val repository: DeviceRepository,
+) {
+    fun remove(deviceId: String) = remove(Identifier(deviceId))
```
Comment (FerranAD) [inline src/main/kotlin/device/application/DeviceRemover.kt:10]: Documentation is missing...

**Review by FerranAD — CHANGES_REQUESTED** [review]
Tells you to add documentation.

File: src/main/kotlin/note/application/NoteDeleter.kt [hunk 2]
```diff
@@ -15,7 +14,8 @@ class NoteDeleter(
      * @throws NonExistentNoteException if the note does not exist or is not saved on the repository.
      * @see Note
      */
-    fun delete(noteId: Identifier) =
```
Comment (Pablito2020) [inline src/main/kotlin/note/application/NoteDeleter.kt:18]: This should be on another different pull request, as stated on #43

File: src/main/kotlin/note/application/NoteSaver.kt [hunk 3]
```diff
@@ -24,13 +24,19 @@ class NoteSaver(
      * @return The saved note instance
      */
     fun save(note: NotePrimitives): Note {
-        val noteIdentifier = Identifier(note.noteId)
+        val noteId = Identifier(note.noteId)
```
Comment (Pablito2020) [inline src/main/kotlin/note/application/NoteSaver.kt:27]: This should be on another different pull request, as stated on #43

File: src/main/kotlin/note/application/NoteSearcher.kt [hunk 4]
```diff
@@ -13,6 +13,6 @@ class NoteSearcher(
      * @see Note
      * @return null or a note instance if it exists
      */
-    fun search(identifier: Identifier) = repository.search(identifier)
```
Comment (Pablito2020) [inline src/main/kotlin/note/application/NoteSearcher.kt:16]: This should be on another different pull request, as stated on #43

**Review by Pablito2020 — CHANGES_REQUESTED** [review]

**Comment by Pablito2020** [comment]
The identifier refactor from note should be on another different pull request, as stated on #43 . Do rebase on the branch only with the remove device changes and then open a pull request again.
